src/internal/connector/exchange/container_resolver.go

```diff
@@ -24,6 +29,18 @@ func (cr *containerResolver) IDToPath(
 	ctx context.Context,
 	folderID string,
 ) (*path.Builder, error) {
+	return cr.idToPath(ctx, folderID, 0)
+}
+
+func (cr *containerResolver) idToPath(
+	ctx context.Context,
+	folderID string,
+	depth int,
+) (*path.Builder, error) {
+	if depth >= maxIterations {
+		return nil, errors.New("path contains cycle or is too tall")
```

Comment:
Is a cyclical path possible?

Reply:
I don't think it should be possible in M365, but this code is generic and just relies on following parentID references. Due to that, it's possible there could be a misconfiguration somewhere that could lead to a cycle and this could would try to resolve it. It's just highly unlikely to encounter that